### PR TITLE
Code review exercise: Kallum 

### PR DESCRIPTION
**Pull request by kallumjones**

### CODE REVIEW

File: exercises/nodejs/src/game.js [hunk 1]
```diff
@@ -1,172 +1,206 @@
...
 
-    currentPlayer += 1
-    if (currentPlayer == players.length) { currentPlayer = 0 }
+    nextPlayer()
     return true
   }
+
+  function nextPlayer() {
+    currentPlayer = (currentPlayer + 1) % playerNames.length;
+  }
```
Comment (chris-cazaly-coop) [inline exercises/nodejs/src/game.js:166]: This `nextPlayer` function makes good use of the `%` operator.
Also it was a good refactor for reducing duplicated code.

File: exercises/nodejs/src/game.js [hunk 2]
```diff
@@ -1,172 +1,206 @@
...
 
   do {
     game.roll(random.range(5) + 1)
 
     if (random.range(9) == 7) {
-      notAWinner = game.wrongAnswer()
+      notAWinner = game.questionAnsweredIncorrectly()
     } else {
-      notAWinner = game.wasCorrectlyAnswered()
+      notAWinner = game.questionAnsweredCorrectly()
```
Comment (chris-cazaly-coop) [inline exercises/nodejs/src/game.js:198]: Good to keep similar functions consistently named! 👍

File: exercises/nodejs/src/game.js [hunk 3]
```diff
@@ -1,172 +1,206 @@
...
+    console.log(questionCategories[category].shift())
   }
 
   this.roll = function (roll) {
-    console.log(players[currentPlayer] + ' is the current player')
+    console.log(playerNames[currentPlayer] + ' is the current player')
     console.log('They have rolled a ' + roll)
 
     if (inPenaltyBox[currentPlayer]) {
       if (roll % 2 != 0) {
```
Comment (chris-cazaly-coop) [inline exercises/nodejs/src/game.js:106]: You could potentially create a method `isOddRoll(roll)` to remove the 'primitive conditional from this statement, and make it read more like a sentence:

```suggestion
      if (isOddRoll(roll)) { ... }
      
      function isOddRoll(roll) {
          return roll % 2 != 0
      }
```

**Review by chris-cazaly-coop — COMMENTED** [review]
🎉🎉 Nice work! 🎉🎉

File: exercises/nodejs/src/game.js (hunk 3 above)
Comment (kallumjones) [reply exercises/nodejs/src/game.js:106]: Thank you, this is a good idea, I will implement locally

File: exercises/nodejs/src/game.js [hunk 4]
```diff
@@ -1,172 +1,206 @@
...
+    return playerNames.length
   }
 
-  var askQuestion = function () {
-    if (currentCategory() == 'Pop') { console.log(popQuestions.shift()) }
-    if (currentCategory() == 'Science') { console.log(scienceQuestions.shift()) }
-    if (currentCategory() == 'Sports') { console.log(sportsQuestions.shift()) }
-    if (currentCategory() == 'Rock') { console.log(rockQuestions.shift()) }
+  var askQuestion = function (category) {
+    console.log(questionCategories[category].shift())
```
[the remark this answers is not in the document]
Comment (kallumjones) [reply exercises/nodejs/src/game.js:98]: Cheers

File: exercises/nodejs/src/game.js [hunk 5]
```diff
@@ -1,172 +1,206 @@
...
-    if (places[currentPlayer] == 8) { return 'Pop' }
-    if (places[currentPlayer] == 1) { return 'Science' }
-    if (places[currentPlayer] == 5) { return 'Science' }
-    if (places[currentPlayer] == 9) { return 'Science' }
-    if (places[currentPlayer] == 2) { return 'Sports' }
-    if (places[currentPlayer] == 6) { return 'Sports' }
-    if (places[currentPlayer] == 10) { return 'Sports' }
+    if (playerPositions[currentPlayer] % 4 == 0) { return 'Pop' }
+    if (playerPositions[currentPlayer] % 4 == 1) { return 'Science' }
+    if (playerPositions[currentPlayer] % 4 == 2) { return 'Sports' }
```
[the remark this answers is not in the document]
Comment (mrmanc) [reply exercises/nodejs/src/game.js:62]: I think there’s an interesting trade-off between the reduction of measured complexity and the fact that it’s now a little harder to understand how the board is laid out due to the use of modulo. You might be interested to hear that there is a way to still very clearly represent the way the board is laid out, and have an even lower complexity score (of 1). Bit of a challenge 😃

[the remark this answers is not in the document]
Comment (M4rt4M) [reply exercises/nodejs/src/game.js:62]: How about this one:
```
  const categories = ['Pop', 'Science', 'Sports', 'Rock']
  var currentCategory = function () { 
      return categories[places[currentPlayer] % categories.length] 
  }
```

**Pull request closed without merging**